src/main/java/net/sf/rails/game/OperatingRound.java

```diff
@@ -1669,9 +1677,11 @@ public boolean layTile(LayTile action) {
                 stl.setExercised();
                 // currentSpecialTileLays.remove(action);
                 log.debug("This was a special tile lay, {} extra", extra ? "" : " not");
-
-            }
-            if (!extra) {
+                if (!extra) {
```

Comment:
That's right. I don't remember why I changed this (i.e. moved `!extra` to within the `stl != null` condition): either because I got an error that later turned out to have a different cause, or because I wanted to emphasize that the `!extra` condition is only meaningful if a SpecialTileLay applies. Anyway, I still think that the new code is a little bit clearer, because you don't have to look back how `extra` was initialized way back.